l10n_br_account_payment_brcobranca/models/account_invoice.py

```diff
@@ -0,0 +1,105 @@
+
+from ..constants.br_cobranca import get_brcobranca_api_url
+
+logger = logging.getLogger(__name__)
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    file_boleto_pdf_id = fields.Many2one(
+        comodel_name="ir.attachment",
+        string="Boleto PDF",
+        ondelete="restrict",
+        copy=False,
+    )
+
+    # Usado para deixar invisivel o botão
+    # Imprimir Boleto, quando não for o caso
+    payment_method_code = fields.Char(related="payment_mode_id.payment_method_id.code")
+
+    def gera_boleto_pdf(self):
+        file_pdf = self.file_boleto_pdf_id
+        self.file_boleto_pdf_id = False
+        file_pdf.unlink()
+
+        receivable_ids = self.mapped("financial_move_line_ids")
+
+        boletos = receivable_ids.send_payment()
+        if not boletos:
+            raise UserError(
+                _(
+                    "It is not possible generated boletos\n"
+                    "Make sure the Invoice are in Confirm state and "
+                    "Payment Mode method are CNAB."
+                )
+            )
+
+        pdf_string = self._get_brcobranca_boleto(boletos)
+
+        inv_number = self.get_invoice_fiscal_number().split("/")[-1].zfill(8)
+        file_name = "boleto_nf-" + inv_number + ".pdf"
+
+        self.file_pdf_id = self.env["ir.attachment"].create(
+            {
+                "name": file_name,
+                "datas_fname": file_name,
+                "res_model": self._name,
+                "res_id": self.id,
+                "datas": base64.b64encode(pdf_string),
+                "mimetype": "application/pdf",
+                "type": "binary",
+            }
+        )
+
+    def _get_brcobranca_boleto(self, boletos):
+
+        content = json.dumps(boletos)
+        f = open(tempfile.mktemp(), "w")
+        f.write(content)
+        f.close()
+        files = {"data": open(f.name, "rb")}
+
+        brcobranca_api_url = get_brcobranca_api_url()
+        brcobranca_service_url = brcobranca_api_url + "/api/boleto/multi"
+        logger.info(
+            "Connecting to %s to get Boleto of invoice %s",
+            brcobranca_service_url,
+            self.name,
+        )
+        res = requests.post(brcobranca_service_url, data={"type": "pdf"}, files=files)
+
+        if str(res.status_code)[0] == "2":
+            pdf_string = res.content
+        else:
+            raise UserError(res.text.encode("utf-8"))
+
+        return pdf_string
+
+    def _target_new_tab(self, attachment_id):
+        if attachment_id:
+            return {
+                "type": "ir.actions.act_url",
+                "url": "/web/content/{id}/{nome}".format(
+                    id=attachment_id.id, nome=attachment_id.name
+                ),
+                "target": "new",
+            }
+
+    def view_boleto_pdf(self):
+        if not self.file_boleto_pdf_id:
+            self.gera_boleto_pdf()
+        return self._target_new_tab(self.file_boleto_pdf_id)
```

Comment:
```suggestion
        return self._target_new_tab(self.file_pdf_id)
```
Estou revisando o código e percebi essa variavel errada.

Reply:
Ola @netosjb obrigado pela revisão, realmente ao clicar no botão deve abrir o pop-up para Vizualizar ou Baixar o Boleto

Reply:
@mbcosta Parabéns pelo trabalho, logo vou estar fazendo mais revisões :D